### PR TITLE
Add summary field on user

### PR DESCRIPTION
**Pull request by medson10**

### CODE REVIEW

File: executable/SummarizeApp.re [hunk 1]
```diff
@@ -4,16 +4,32 @@ open AppSchema;
...
 
-let users = [
-  {id: 1, name: "Alice", role: Admin},
-  {id: 2, name: "Bob", role: User},
-];
+let mockedUser: user = {
+  id: 1,
+  name: "Alice",
+  role: Admin,
+  summary: {
```
Comment (fakenickels) [inline executable/SummarizeApp.re]: this actually won't come from the user model, it'll be aggregated just in time (commenting below)

File: executable/SummarizeApp.re [hunk 2]
```diff
@@ -46,6 +102,10 @@ let user =
         field("role", ~args=Arg.[], ~typ=non_null(role), ~resolve=(info, p) =>
           p.role
         ),
+        field(
+          "summary", ~args=Arg.[], ~typ=non_null(summary), ~resolve=(info, p) =>
+          p.summary
```
Comment (fakenickels) [inline executable/SummarizeApp.re]: here we'll do the aggregation and we'll return the `let summary = { unread: result, etc }`

File: executable/SummarizeApp.re [hunk 3]
```diff
@@ -79,7 +132,11 @@ module Graphql_cohttp_lwt =
 let _ =
   Lwt_main.run(
     {
-      let callback = Graphql_cohttp_lwt.make_callback(_req => Context.{user: None}, schema);
+      let callback =
+        Graphql_cohttp_lwt.make_callback(
+          _req => Context.{user: None},
```
Comment (fakenickels) [inline executable/SummarizeApp.re]: you can actually put the mocked user here

File: executable/SummarizeApp.re [hunk 4]
```diff
@@ -54,15 +114,8 @@ let schema =
...
-        io_field(
-          "users",
-          ~typ=non_null(list(non_null(user))),
-          ~args=Arg.[],
-          ~resolve=(_info, ()) =>
-          Lwt.return(Ok(users))
-        ),
         io_field("currentUser", ~typ=user, ~args=Arg.[], ~resolve=(_info, ()) =>
-          Lwt.return(Ok(None))
+          Lwt.return(Ok(Some(mockedUser)))
```
Comment (fakenickels) [inline executable/SummarizeApp.re]: return the user from context here

File: package.json [hunk 5]
```diff
@@ -31,6 +31,7 @@
     }
   },
   "scripts": {
+    "start": "esy b && esy x SummarizeApp.exe",
```
Comment (fakenickels) [inline package.json:34]: good one

File: schema/Context.re [hunk 6]
```diff
@@ -7,6 +13,7 @@ type user = {
   id: int,
   name: string,
   role,
+  summary,
```
Comment (fakenickels) [inline schema/Context.re]: as I said above, remove the `summary` field from the user record

File: schema/User.re [hunk 7]
```diff
@@ -0,0 +1,16 @@
+type role =
```
Comment (fakenickels) [inline schema/User.re:1]: nice we can even experiment after having Model + entity GraphQL schema in one file after

**Review by fakenickels — APPROVED** [review]

File: schema/User.re (hunk 7 above)
Comment (medson10) [reply schema/User.re:1]: yeah nice one, probably will be easier to pass types reference, actually by what I was seeing is how they do with elixir

Comment (fakenickels) [reply schema/User.re:1]: makes totally sense indeed

**Comment by guilhermedecampo** [comment]
🚀

**Pull request merged**